src/modules/api/graphql/resolvers.ts

```diff
@@ -11,6 +10,46 @@ export const DebugResolvers = {
   }
 }
 
+function tryJSONParse(str): string {
+  try {
+      return JSON.parse(str);
+  } catch (e) {
+      return str;
+  }
+}
+
+const getRelevantContentOfIpfsHash = async (ipfsHash: string) => {
+  let content = await appContainer.self.ipfs.dag.get(CID.parse(ipfsHash)) as any;
+
+  // determine if ipfs hash resembles a regular ipfs object and not an ipfs cbor object 
+  if ('Links' in content.value && 'Data' in content.value) {
+
+    const stream = appContainer.self.ipfs.cat(CID.parse(ipfsHash));
+    const chunks = [];
+
+    for await (const chunk of stream) {
+      chunks.push(chunk);
+    }
+
+    const buffer = Buffer.concat(chunks);
+    let dataRaw = buffer.toString(); 
+
+    content = tryJSONParse(dataRaw);
+  } else {
+    content = content.value
+  }
+
+  if (typeof content === 'object') {
```

Comment:
Might need to detect any infinite recursion as contents of a CBOR JSON could end up reference its own CID

Reply:
great catch! implemented ✌️